lib/licensed/sources/gradle.rb

```diff
@@ -9,127 +9,156 @@ module Licensed
   module Sources
     class Gradle < Source
 
-      DEFAULT_CONFIGURATIONS   = ["runtime", "runtimeClasspath"]
-      GRADLE_LICENSES_PATH     = ".gradle-licenses"
-      GRADLE_LICENSES_CSV_NAME = "licenses.csv"
+      DEFAULT_CONFIGURATIONS   = ["runtime", "runtimeClasspath"].freeze
+      GRADLE_LICENSES_PATH     = ".gradle-licenses".freeze
 
-      # Returns the configurations to include in license generation.
-      # Defaults to ["runtime", "runtimeClasspath"]
-      def configurations
-        @configurations ||= begin
-          if configurations = config.dig("gradle", "configurations")
-            Array(configurations)
-          else
-            DEFAULT_CONFIGURATIONS
+      class Dependency < Licensed::Dependency
+        GRADLE_LICENSES_CSV_NAME = "licenses.csv".freeze
+
+        class << self
+          # Returns a key to uniquely identify a name and version in the obtained CSV content
+          def csv_key(name:, version:)
+            "#{name}-#{version}"
+          end
+
+          # Loads and caches license report CSV data as a hash of :name-:version => :url pairs
+          #
+          # executable     - The gradle executable to run to generate the license report
+          # configurations - The gradle configurations to generate license report for
+          #
+          # Returns a hash of dependency identifiers to their license content URL
+          def load_csv(executable, configurations)
+            @csv ||= begin
+              Licensed::Sources::Gradle.gradle_command("generateLicenseReport", executable: executable, configurations: configurations)
+              CSV.foreach(File.join(path, GRADLE_LICENSES_CSV_NAME), headers: true).each_with_object({}) do |row, hsh|
+                name, _, version = row["artifact"].rpartition(":")
+                key = csv_key(name: name, version: version)
+                hsh[key] = row["moduleLicenseUrl"]
+              end
+            end
           end
+
+          # Returns the cached url for the given dependency
+          def url_for(dependency)
+            @csv[csv_key(name: dependency.name, version: dependency.version)]
+          end
+
+          # Cache and return the results of getting the license content.
+          def license(url)
+            (@licenses ||= {})[url] ||= Net::HTTP.get(uri)
+          end
+        end
+
+        def initialize(name:, version:, path:, executable: ,configurations:, metadata: {})
+          @configuration = configurations
+          @executable = executable
+          super(name: name, version: version, path: path, metadata: metadata)
+        end
+
+        # Returns whether the dependency content exists
+        def exist?
+          # shouldn't force network connections just to check if content exists
+          # only check that the path is not empty
+          !path.to_s.empty?
+        end
+
+        # Returns a Licensee::ProjectFiles::LicenseFile for the dependency
+        def project_files
```

Comment:
@jonabc Hmm, I'm not sure how this works? I see `lib/licensed/sources/manifest.rb` also contains this method but I don't see it called anywhere? Am I missing something here?

Reply:
@dbussink with licensed v2, `Licensed::Dependency` inherits from `Licensee::FSProject` to be able to directly provide additional content to licensee for cases such as this or the manifest source.

One way to provide additional content is by overriding this method.  The base method is  [`Licensee::Projects::Project#project_files`](https://github.com/licensee/licensee/blob/da21bba0352f8086920ec59e61c8dbd93f4f0e6d/lib/licensee/projects/project.rb#L148), which is used internal to licensee as the source of license file content to evaluate.